cmd/export/main.go

```diff
@@ -17,37 +17,102 @@ package main
 
 import (
 	"context"
-	"log"
+	"fmt"
 	"net/http"
+	"time"
 
 	"go.opencensus.io/plugin/ochttp"
 
 	"github.com/google/exposure-notifications-server/internal/export"
 	"github.com/google/exposure-notifications-server/internal/logging"
 	_ "github.com/google/exposure-notifications-server/internal/observability"
+	"github.com/google/exposure-notifications-server/internal/serverenv"
 	"github.com/google/exposure-notifications-server/internal/setup"
 )
 
 func main() {
 	ctx := context.Background()
 	logger := logging.FromContext(ctx)
 
+	if err := realMain(); err != nil {
+		logger.Fatal(err)
+	}
+}
+
+func realMain() error {
+	ctx := context.Background()
+
+	config, env, closer, err := doSetup()
+	if err != nil {
+		return fmt.Errorf("setup: %w", err)
+	}
+	defer closer()
+
+	server, err := NewServer(ctx, config, env)
+	if err != nil {
+		return fmt.Errorf("newserver: %w", err)
+	}
+
+	go server.Run()
```

Comment:
I'm not really sure what the best design is here. Cloud Run seems to send SIGKILL (which we can't trap), but that also means we have no opportunity to cleanup connections.

Reply:
correct - I've already relayed this feedback that graceful shutdown signals are needed.